### DISCUSSION TITLE
OAuth2 and Bearer Authorization

### DISCUSSION THREAD

**Discussion by credas-dev** [jellydn/next-swagger-doc#655] [Q&A]
Is there way to configure swagger for different types of security?

I found a discussion [here](https://stackoverflow.com/questions/50736784/how-to-authorise-swagger-jsdoc-with-jwt-token) that we could configure security in the format below.

```
const swaggerOptions = {
  swaggerDefinition: {
    openapi: '3.0.1', // YOU NEED THIS
    info: {
      title: 'Your API title',
      version: '1.0.0',
      description: 'Your API description'
    },
    basePath: '/',
    components: {
      securitySchemes: {
        bearerAuth: {
          type: 'http',
          scheme: 'bearer',
          bearerFormat: 'JWT',
        }
      }
    },
    security: [{
      bearerAuth: []
    }]
  },
  apis: ['/some/path.js|yml'],
};

```
I'm using NextJS 12 and configuring my docs using the code below.

```
import { withSwagger } from "next-swagger-doc";

const swaggerHandler = withSwagger({
  definition: {
    openapi: "3.0.0",
    info: {
      title: "OTLS-API",
      version: "1.39.0",
    },
  },
  apiFolder: "pages/api",
});
export default swaggerHandler();

```
I wasn't able to configure a security parameter in the definition. Is there an example of how I can do this?

**Comment by jellydn** [comment]
Hi @credas-dev 
I've tried this yet but I think this should work.

```
import { withSwagger } from "next-swagger-doc";

const swaggerHandler = withSwagger({
  definition: {
    openapi: "3.0.0",
    info: {
      title: "OTLS-API",
      version: "1.39.0",
    },
    // Add Auth 
     components: {
      securitySchemes: {
        bearerAuth: {
          type: 'http',
          scheme: 'bearer',
          bearerFormat: 'JWT',
        }
      }
    },
    security: [{
      bearerAuth: []
    }]
  },
  apiFolder: "pages/api",
});
```

**Reply by jmsolutionsgroup** [reply]
@jellydn 
I created a sample [repo ](https://github.com/uchetfield/next-swagger-doc-auth)using a modified version of the sample that is provided with version v.0.3.6.  

I'm still not getting the prompt or an area for a bearer token.   Can you see if something is wrong?

**Reply by jellydn** [reply]
Hi @uchetfield I've added a example on this repo for your question. 
<img width="1722" alt="image" src="https://user-images.githubusercontent.com/870029/229270474-2372bab9-fffa-42ee-af9b-602feee8c30d.png">
 Let me know if you have any questions.

**Reply by jellydn** [reply]
You could refer to this commit for usage https://github.com/jellydn/next-swagger-doc/commit/1a5a08e2ade6f636b29d37acfbbec6596b79274e
Demo link: https://next-swagger-doc-demo.productsway.com/api-doc

**Reply by jmsolutionsgroup** [reply]
Thank you for the update.  What would be the correct way to set the oauth2RedirectUrl?

![image](https://github.com/jellydn/next-swagger-doc/assets/5488668/d277a434-1388-4611-9388-7082f246141b)
